src/bvals/comms/boundary_communication.cpp

```diff
@@ -252,16 +252,6 @@ TaskStatus SetBounds(std::shared_ptr<MeshData<Real>> &md) {
     auto pmb = md->GetBlockData(0)->GetBlockPointer();
     StateDescriptor *resolved_packages = pmb->resolved_packages.get();
     refinement::Restrict(resolved_packages, cache, pmb->cellbounds, pmb->c_cellbounds);
-
-    // Apply coarse boundary conditions
-    for (int block = 0; block < md->NumBlocks(); ++block) {
-      ApplyBoundaryConditionsOnCoarseOrFine(md->GetBlockData(block), true);
-    }
-
-    // Prolongate from coarse buffer
-    refinement::Prolongate(resolved_packages, cache, pmb->cellbounds, pmb->c_cellbounds);
-    refinement::ProlongateInternal(resolved_packages, cache, pmb->cellbounds,
-                                   pmb->c_cellbounds);
```

Comment:
I guess these need to be split out into tasks again because of the issue with local vs nonlocal send/receive?

Reply:
Yeah, you can't prolongate until you have received all data. As it was before, I think it would be safe with local and nonlocal, but you would prolongate twice and only the second time through would be correct. I think it is just easier to think through and less error prone to keep these separate.

Reply:
:+1: